File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let totalSum = nums.reduce((acc,cur)=>acc+cur,0);
    let leftSum = 0;
    for(let i = 0 ; i < nums.length ; i++){
        totalSum -= nums[i];
        if(totalSum === leftSum)
            return i;
            leftSum += nums[i]
    }
    return -1;
    
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total_sum = sum(nums)
        left_sum = 0
        for i in range(len(nums)):
            total_sum -= nums[i]
            if total_sum == left_sum:
                return i
            else:
                left_sum += nums[i]
        return -1
        